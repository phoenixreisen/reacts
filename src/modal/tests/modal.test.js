/**
 * @jest-environment jsdom
 */
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import modalSizes from '../modal.sizes';
import Modal from '../modal.r';
import React from 'react';

describe('Modal should', () => {

    test('render correctly', () => {
        const state = {
            show: true
        };
        const component = renderer.create(
            <Modal
                key="modal"
                withCloseText={false}
                title={"Awesome Modal"}
                content={<div>Content</div>}
                footer={<div><button>Schließen</button></div>}
                toggle={() => state.show = false}
            />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('call toggle function', () => {
        const state = {
            show: true
        }
        const component = render(
            <Modal
                key="modal"
                withCloseText={true}
                title={"Awesome Modal"}
                content={<div>Content</div>}
                footer={<div><button>Toggle</button></div>}
                toggle={() => state.show = false}
            />
        );
        expect(screen.getByText('Toggle')).not.toBeNull();
        expect(screen.getByText('Content')).not.toBeNull();
        expect(screen.getByText('schließen')).not.toBeNull();
        expect(screen.getByText('Awesome Modal')).not.toBeNull();

        expect(state.show).toBe(true);
        fireEvent.click(document.querySelector('a.modal__toggle'));
        expect(state.show).toBe(false);
    });

    test('not render without content', () => {
        const out = console.error;
        console.error = () => {};
        let error = false;
        try {
            render(<Modal />);
        } catch(e) {
            error = true;
        }
        expect(error).toBe(true);
        console.error = out;
    });

    test('not render with an unknown size prop', () => {
        const out = console.error;
        console.error = () => {};

        // unknown size
        let error = false;
        try {
            render(<Modal size={"90x90"} title="Title" content={<div>Hallo</div>} />);
        } catch(e) {
            error = true;
        }
        expect(error).toBe(true);

        // known size
        error = false;
        try {
            render(<Modal size={modalSizes.s7590} title="Title" content={<div>Hallo</div>} />);
        } catch(e) {
            console.log(e);
            error = true;
        }
        expect(error).toBe(false);
        console.error = out;
    });
});
