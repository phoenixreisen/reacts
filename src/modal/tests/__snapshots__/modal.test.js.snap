// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
[
  <article
    className="modal modal--visible "
  >
    <div
      className="modal__header"
    >
      <span
        className="modal__headline"
      >
        Awesome Modal
      </span>
      <a
        className="modal__toggle"
        href="#"
        onClick={[Function]}
      >
        <i
          className="fas fa-times-circle"
        />
      </a>
    </div>
    <div
      className="modal__content"
    >
      <div>
        Content
      </div>
    </div>
    <div
      className="modal__footer tr"
    >
      <div>
        <button>
          Schlie√üen
        </button>
      </div>
    </div>
  </article>,
  ",",
  <article
    className="modal__bg"
  />,
  ",",
]
`;
